src/core/lib/promise/activity.cc

```diff
@@ -0,0 +1,112 @@
+// Copyright 2021 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <grpc/impl/codegen/port_platform.h>
+
+#include "src/core/lib/gprpp/atomic_utils.h"
+#include "src/core/lib/promise/activity.h"
+
+namespace grpc_core {
+
+///////////////////////////////////////////////////////////////////////////////
+// GLOBALS
+
+thread_local Activity* Activity::g_current_activity_ = nullptr;
+Waker::Unwakeable Waker::unwakeable_;
```

Comment:
So it's possible to fix this... and there are two options:

1. add an extra if to every path that's using wakeables to verify it's been dropped or not (this is likely a measurable runtime cost)
2. change Wakeable to something like:
```
struct Wakeable {
  WakeableVTable* const vtable;
};
struct WakeableVTable {
  void (*wake)(Wakeable*);
  void (*drop)(Wakeable*);
};
```
Which will have the same runtime cost, but cost an extra pointer per runtime object - and we need to keep this object fairly lean.

I think either cost is too high in this case, and so we should take this global.

Reply:
(if there's an option that does not increase runtime cost that I'm not seeing, please lmk too)

Reply:
Understood. I don't have a better solution to offer. I think we'll need to be careful of anything else we introduce with static storage duration that wants to rely on a `Waker`, since we could run into nondeterministic ordering of destruction.